Highlight names of modified properties in the inspector

Calinou opened the pull request:
This is more in line with how other software denotes modified properties in inspector-like controls.

## Preview

### Standard node

| Dark theme | Light theme |
|-|-|
| ![2022-06-19_07 02 12](https://user-images.githubusercontent.com/180032/174466837-2ad63f26-a96d-474e-8fad-ee86cd280b4c.png) | ![2022-06-19_07 01 41](https://user-images.githubusercontent.com/180032/174466836-a3cd1750-1b57-4ebd-83aa-eb9644969ea7.png) |

### Node edited via Editable Children

| Dark theme | Light theme |
|-|-|
| ![2022-06-19_07 05 06](https://user-images.githubusercontent.com/180032/174466838-0723e0fb-4758-494e-a452-f5adcd35ea2e.png) | ![2022-06-19_07 05 51](https://user-images.githubusercontent.com/180032/174466840-916597ba-2510-4232-bff2-eef9e1ebcf43.png) |

groud commented:
It kind of looks disturbing IMHO.  I would say the revert icon is enough for me.
But I am not strongly against it, I believe we need more opinions.

capnm commented:
> It kind of looks disturbing IMHO. 

:+1:

Phischermen commented:
I like that this will make Godot more consistent with other programs. Previously the revert icon was the only way to identify modified properties, but since that is no longer the case then maybe the button can be removed and its functionality put in a menu you can access by right clicking a property.

dreamsComeTrue commented:
One suggestion @Calinou - can you make that color - configurable? Nice job BTW

Calinou commented:
@dreamsComeTrue I don't see much point in making it configurable. We have a lot of Editor Settings related to customization already.

samdze commented:
I'm ok with this change, but I agree that this may not be enough to properly highlight a modified property.
As @Phischermen said, the revert icon doesn't appear only on modifier properties, but also on non-null resource properties for example.
Why not use the editor highlight color to do something like this?

![immagine](https://user-images.githubusercontent.com/19392104/89779038-d7840500-db0e-11ea-927d-49f550336915.png)

groud commented:
To be honest, I don't understand why you would need to highlight that much modified properties. It's not because you set a property as different to the default value that they need to be standing out like if they were significantly more important from the other properties.

I'd really prefer we keep the effect subtle enough so that it does not disturb your eye too much when you search for other properties. I'd prefer, for example, changing the property's value color to blue when it's not the default one, instead of cheging the property title to bold. (I'd remove the blue color for ticked checkboxes in such case then)

samdze commented:
Personally I do feel that modified properties are significantly more important than the others.
They are what set apart the instanced scenes from the original (not considering additional nodes), plus they indicate what properties will stay the way they are if those are modified in the original scene.
I worked on several big projects on Unity, and clearly seeing what properties differ from the prefab helped me a lot.

Phischermen commented:
@groud Is this what you had in mind?
![GodotInspectorAccent](https://user-images.githubusercontent.com/46539626/89830511-259a0800-db11-11ea-8699-78ca787e7253.png)
Comparing @groud and @samdze 's proposals to the original, I can see that color contributes better to identifiablity than bold text, especially if the color could be configured so even color-blind people could perceive it.

Calinou commented:
> especially if the color could be configured so even color-blind people could perceive it.

If we decide to change the color instead of the font weight, it's just matter of changing the accent color in the editor theme settings :slightly_smiling_face:

hoontee commented:
Yes, a color change / color indicator would make it look less unnatural IMO.

groud commented:
@Phischermen I was thinking about changing the values color, not the property name. But yes, the idea is there.

ShivamMukherjee commented:
I believe on a dark background (which most Godot users have by default) the additional font weight will be hard to detect. A colour change (additionally or alone) must be present.

akien-mga commented:
I'm also not convinced by the font weight change.

Using the accent color might work, though I'd tend to do it for the modified value and not the modified property name, as suggested by @groud.

samdze commented:
I don't think the change in the value's color would be the better solution.

This way every property editor would have to adapt and use the highlight color in a slightly different way to make it clear that it has been changed.
Integers, floats, ranges, strings, colors, resources, curves, bitmasks, etc... each would have to support a way to show it has been modified, and consider that this should be easily supported by custom property editors too.

Changing the property name in some way or adding a mark to the property slot is a universal and consistent solution.

akien-mga commented:
So what's the next iteration of this PR? I think we agree that the original bold proposal wouldn't be the best option. Should we try one of the other mockups? How does it play with the now color-highlighted subresource edition?

Calinou commented:
@akien-mga I'll try to change the property name's color a bit instead of changing its font weight. Maybe I can also draw a subtle line on the left of the property name.

YeldhamDev commented:
I would personally go for the color change, since adding anything to the name would take precious horizontal space.

Calinou committed:
Highlight names of modified properties in the inspector

This is more in line with how other software denotes modified properties
in inspector-like controls.

Calinou commented:
Redone to change the color of property names instead of making them bold.

akien-mga commented:
Superseded by #63429.

ajreckof commented:
@akien-mga  I'm not sure if the improvement in this PR should be added or not but it does not seem like the PR you linked to supersedes this one as the one you link is to add visibility of the to the groups that have been edited while this to add visibility to the properties themselves.

Calinou commented:
> @akien-mga I'm not sure if the improvement in this PR should be added or not but it does not seem like the PR you linked to supersedes this one as the one you link is to add visibility of the to the groups that have been edited while this to add visibility to the properties themselves.

That PR is technically different, but we discussed this PR in a meeting and rejected it anyway.

samdze commented:
So, we're leaving it just to the revert icon the job of indicating the presence of an edited property?

I prefer a more decise way of displaying them, also because the revert icon doesn't appear only on modified properties, but also on non-null resource properties (not checked if still the case in 4.x) for example.

akien-mga commented:
> I prefer a more decise way of displaying them, also because the revert icon doesn't appear only on modified properties, but also on non-null resource properties (not checked if still the case in 4.x) for example.

Non-null resource properties *are* modified by definition, no? Or what do you have in mind exactly?

samdze commented:
> Non-null resource properties are modified by definition, no? Or what do you have in mind exactly?ù

In simple cases, yes, but not in scenes inheriting other scenes.

akien-mga commented:
It's the same. In inherited scenes, there should be a revert icon if the value differs from the scene it inherits from.

Just tested in 3.5 and it seems to work as intended:

- Base scene with non-default material:

![image](https://user-images.githubusercontent.com/4701338/181616384-0f34756e-e61b-40b2-adce-bcaea7e61150.png)

- Inherited scene with no change:

![image](https://user-images.githubusercontent.com/4701338/181616525-8e57f120-e560-430a-9893-6e90c14d75ab.png)

- Inherited scene with overridden material:

![image](https://user-images.githubusercontent.com/4701338/181616634-a1cd7907-90f2-479f-990c-f24a23db4969.png)

samdze commented:
Oh, it has been fixed then! Time ago the revert icon appeared no matter what. Nevermind!

KoBeWi commented:
btw the property highlight uses the same logic as revert arrow, so it wouldn't make it any better.
